yarn-project/aztec-nr/aztec/src/note/note_viewer_options.nr

```diff
@@ -21,9 +21,11 @@ impl<Note, N> NoteViewerOptions<Note, N> {
             offset: 0,
         }
     }
-
-    pub fn select(&mut self, field_index: u8, value: Field) -> Self {
-        self.selects.push(Option::some(Select::new(field_index, value)));
+       
+    // This method adds a `Select` criterion to the options.
+    // It takes a field_index indicating which field to select and a value representing the specific value to match in that field.
+    pub fn select(&mut self, field_index: u8, value: Field, comparator: Option<u3>) -> Self {
```

Comment:
Would put here a comment listing out the encoding of comparators (or point to the comparator enum). All the users would be looking for this when reading this method.